CONTRIBUTING.md

```diff
@@ -50,3 +63,7 @@ For example:
 niv update nixpkgs -r 4e329926df7ee5fa49929a83d31ee7d541f8b45c
 niv update nixpkgs -v 21.11.337905.902d91def1e
 ```
+
+
+[Agda]: https://wiki.portal.chalmers.se/agda/pmwiki.php
+[Agda standard library style guide]: https://github.com/agda/agda-stdlib/blob/master/notes/style-guide.md
```

Comment:
What happened to the `nix-build -A agdaLedger` command?  What was that for?

Tbh, I don't yet fully grasp what the nix-build commands do in each case... but I'll have a look at the *.nix files and figure it out.  (Still learning nix/nixos so this is probably good exercise for me.)

Reply:
`nix-build -A X` looks into `default.nix` and builds derivation `X`. If you look at the bottom of ours, it has the two lines `ledger = specsDerivations ...` and `midnight = specsDerivations ...`. `specsDerivations` is defined as `{ ... }: let ... in { executableSpecSrc = ...; executableSpec = ...; docs = ...; }` so it returns a set with those three elements. Each of those exist for `ledger` and `midnight`, so there are six derivations in total. So you can do something like `nix-build -A ledger.docs` to build that specific derivation, but also `nix-build -A ledger` to build all three for ledger.

Additionally, above `specsDerivation` there are three other derivations, `agda`, `agda2` (that one is a hack for working on stdlib-meta) and `agdaLedger`. The two Agdas just build Agda with libraries and `agdaLedger` is the Agda package of this repo. Building that type checks all of our Agda code and generates a package that is used by the other derivations. If you just run `nix-build`, it'll build all derivations at the top level, so exactly those three. So you never need to run `nix-build -A agdaLedger`. In the past, some of the other derivations also used to be at the top level which was slow, so it was useful to single it out, but the two Agdas should always be cached.

Reply:
Nice explanation.  Thank you.  I don't suppose we should add any of that information to `Contributing.md`, should we?  Or should we assume those who intend to contribute will know enough about nix that they can deduce all that from `default.nix`?

Reply:
I don't think we should try to be a documentation for nix (or the parts of it we use), but I also wouldn't assume that people know nix. We could link to the nix documentation, but that's not actually useful for beginners I think. I have no idea what's best here.